### ISSUE TITLE
Cache scanned products locally so that they load faster

### ISSUE THREAD

**Issue by teolemon** [openfoodfacts/openfoodfacts-ios#882]
### What

- Cache scanned products locally so that they load faster, even if the phone is offline, and even if the server is down or slow at that peculiar moment.
- Refresh should be done if network is available, after showing the cached version.

### Why 
- Slow or no network in supermarkets
- Data is still expensive or unavailable in many areas

**Comment by jordan-furr** [comment]
Is anyone working on this?

**Comment by teolemon** [comment]
@jordan-furr no, not at the moment. You're very welcome to assign it to yourself if you wish.
